manticore/platforms/evm.py

```diff
@@ -500,35 +689,244 @@ class EVMDecoder(object):
                 0xff: ('SELFDESTRUCT', 0, 1, 0, 5000, 'Halt execution and register account for later deletion.')
             }
 
+    @staticmethod
+    @memoized
+    def _get_reverse_table():
+        ''' Build an internal table used in the assembler '''
+        reverse_table = {}
+        for (opcode, (name, immediate_operand_size, pops, pushes, gas, description)) in EVMAssembler._table.items():
+            mnemonic = name
+            if name == 'PUSH':
+                mnemonic = '%s%d'%(name, (opcode&0x1f) + 1)
+            elif name in ('SWAP', 'LOG'):
+                mnemonic = '%s%d'%(name, (opcode&0xf) + 1)
+
+            reverse_table[mnemonic] = opcode, name, immediate_operand_size, pops, pushes, gas, description
+        return reverse_table
+
+    @staticmethod
+    def assemble_one(assembler, offset=0):
+        ''' Assemble one EVM instruction from its textual representation. 
+            
+            :param assembler: assembler code for one instruction
+            :param offset: offset of the instruction in the bytecode (optional)
+
+            :return: An Instruction object
+
+            Example use::
+
+                >>> print evm.EVMAssembler.encode_one('LT')
+            
+
+        '''
+        try:
+            _reverse_table = EVMAssembler._get_reverse_table()
+            assembler = assembler.strip().split(' ')
+            opcode, name, operand_size, pops, pushes, gas, description = _reverse_table[assembler[0].upper()]
+            if operand_size > 0:
+                assert len(assembler) == 2
+                operand = int(assembler[1],0)
+            else:
+                assert len(assembler) == 1
+                operand = None
+
+            return EVMAssembler.Instruction(opcode, name, operand_size, pops, pushes, gas, description, operand=operand, offset=offset)
+        except:
+            raise Exception("Something wron at offset %d"%offset)
 
     @staticmethod
-    def decode_one(bytecode):
+    def assemble_all(assembler, offset=0):
+        ''' Assemble a sequence of textual representation of EVM instructions 
+
+            :param assembler: assembler code for any number of instructions
+            :param offset: offset of the first instruction in the bytecode(optional)
+
+            :return: An generator of Instruction objects
```

Comment:
We can do some really fancy stuff and get links in the docs if we want

see: https://github.com/trailofbits/manticore/blob/master/manticore/manticore.py#L450